src/logging.jl

```diff
@@ -0,0 +1,204 @@
+        col = :light_yellow
+    elseif string(lvl)=="Info"
+        col = :cyan
+    else
+        col = :default
+    end
+
+    printstyled(logger.io, "┌ ", _mod, " | ", lvl, ": "; bold=true, color=col)
+    printstyled(logger.io, msg1, "\n")
+    for msg in rest
+        println(logger.io, "│ ", msg)
+    end
+    for (key, val) in kwargs
+        key === :maxlog && continue
+        println(logger.io, "│   ", key, " = ", val)
+    end
+    println(logger.io, "└ @ ", _mod, " ", file, ":", line)
+
+    if string(lvl) in ["Warn","Error"]
+        if string(lvl) ∉ keys(logger.d) # key doesnt exists
+            logger.d[string(lvl)] = Dict()
+        end
+
+        # Does the key for file exist?
+        if Sys.iswindows() #windows
+            splitter = "\\"
+        else # unix/mac
+            splitter = "/"
+        end
+        
+        splt = split(file, splitter)
+        f = join([splt[end-1], splt[end], line], "_")
+
+        if f ∉ keys(logger.d[string(lvl)]) #file name doesnt exists
+            logger.d[string(lvl)][f] = Any[]
+        end
+
+        push!(logger.d[string(lvl)][f], msg)
+    end
+end
+
+"""
+    handle_errors(e::E, stacktrace::V) where {
+		E <: Exception,
+		V <: Vector
+	}
+
+Creates a results dictionary in case of an error from REopt.jl with Warnings and Errors from logREopt.d. The unhandled error+stacktrace is returned to the user.
+"""
+function handle_errors(e::E, stacktrace::V) where {
+	E <: Exception,
+	V <: Vector
+	}
+
+	results = Dict(
+		"Messages"=>Dict(),
+		"status"=>"error"
+	)
+
+	results["Messages"] = logger_to_dict()
+
+	push!(results["Messages"]["errors"], (string(e),string.(stacktrace)))
+	return results
+end
+
+"""
+    handle_errors()
+
+Creates a results dictionary in case of a handled error from REopt.jl with Warnings and Errors from logREopt.d, which is returned to the user.
+"""
+function handle_errors()
+
+	results = Dict(
+		"Messages"=>Dict(),
+		"status"=>"error"
+	)
+
+	results["Messages"] = logger_to_dict()
+
+	return results
+end
+
+"""
+	logger_to_dict()
+
+The purpose of this function is to extract warnings and errors from REopt logger and package them in a dictionary which can be returned to the user as-is.
+"""
+function logger_to_dict()
+
+	d = Dict()
+	d["warnings"] = []
+	d["errors"] = []
+
+	if "Warn" in keys(logREopt.d)
+		for (keys,values) in logREopt.d["Warn"]
+			push!(d["warnings"], (keys, values))
+		end
+	end
+
+	if "Error" in keys(logREopt.d)
+		for (keys,values) in logREopt.d["Error"]
+			push!(d["errors"], (keys, values))
+		end
+	end
+
+	return d
+end
+
+"""
+	instantiate_logger()
+
+Instantiate a global logger of type REoptLogger and set it to global logger for downstream processing.
+"""
+function instantiate_logger()
+
+	if !isa(global_logger(), REoptLogger)
+		global logREopt = REoptLogger()
+		global_logger(logREopt)
+		@debug "Created custom REopt Logger"
+	else
+		@debug "Already REoptLogger"
+		@debug typeof(global_logger())
```

Comment:
I'm deleting this line because global_logger must be a REoptLogger based on if failing

Reply:
Sounds good!